test/unit/uk/gov/hmrc/apinotificationqueue/controller/QueueControllerSpec.scala

```diff
@@ -51,43 +51,70 @@ class QueueControllerSpec extends UnitSpec with MockitoSugar with WithFakeApplic
     }
 
     val mockQueueService = mock[QueueService]
-    val queueController = new QueueController(mockQueueService, new StaticIDGenerator)
+    val mockFieldsService = mock[ApiSubscriptionFieldsService]
+    val queueController = new QueueController(mockQueueService, mockFieldsService, new StaticIDGenerator)
   }
 
   "POST /queue" should {
-    "throw exception when then X-Client-ID header is not sent in the request" in new Setup {
-      intercept[BadRequestException] {
-        await(queueController.save()(FakeRequest(POST, "/queue")))
-      }
+    "return 400 when then X-Client-ID header is not sent in the request" in new Setup {
+      val result = await(queueController.save()(FakeRequest(POST, "/queue")))
+
+      status(result) shouldBe Status.BAD_REQUEST
     }
 
-    "throw exception if no payload" in new Setup {
-      val request = FakeRequest(POST, "/queue", Headers(CLIENT_ID_HEADER_NAME -> clientId), AnyContentAsEmpty)
-      intercept[BadRequestException] {
-        await(queueController.save()(request))
-      }
+    "return 400 if api-subscription-fields not there in fieldsService" in new Setup {
```

Comment:
The title of this test is not very clear to me.
Perhaps you could say 
```
return 400 when the `fieldsId` does not exist in the `api-subscription-fields` service
```
(if this was the scenario you wanted to cover with your test)